### PR TITLE
Add GetCountAsync method

### PR DESCRIPTION
**Pull request by realLiangshiwei**

### CODE REVIEW

**Review by maliming — APPROVED** [review]

File: modules/identity/src/Volo.Abp.Identity.MongoDB/Volo/Abp/Identity/MongoDB/MongoIdentityUserRepository.cs [hunk 1]
```diff
@@ -269,6 +269,25 @@ public MongoIdentityUserRepository(IMongoDbContextProvider<IAbpIdentityMongoDbCo
             .LongCountAsync(GetCancellationToken(cancellationToken));
     }
 
+    public async Task<List<RoleWithUserCount>> GetCountAsync(Guid[] roleIds, CancellationToken cancellationToken = default)
+    {
+        var users = await (await GetMongoQueryableAsync(cancellationToken))
```
Comment (hikalkan) [inline modules/identity/src/Volo.Abp.Identity.MongoDB/Volo/Abp/Identity/MongoDB/MongoIdentityUserRepository.cs:274]: This won't be efficient. You are getting all users from database. What if there are thousands of users? I believe we can get roles with counts with a single query. Isn't that possible?

File: modules/identity/src/Volo.Abp.Identity.EntityFrameworkCore/Volo/Abp/Identity/EntityFrameworkCore/EfCoreIdentityUserRepository.cs [hunk 2]
```diff
@@ -263,6 +263,25 @@ select ouRole.RoleId
             .LongCountAsync(GetCancellationToken(cancellationToken));
     }
 
+    public async Task<List<RoleWithUserCount>> GetCountAsync(Guid[] roleIds, CancellationToken cancellationToken = default)
```
Comment (hikalkan) [inline modules/identity/src/Volo.Abp.Identity.EntityFrameworkCore/Volo/Abp/Identity/EntityFrameworkCore/EfCoreIdentityUserRepository.cs:266]: This method is not performant. Should be revisited.

Comment (realLiangshiwei) [reply modules/identity/src/Volo.Abp.Identity.EntityFrameworkCore/Volo/Abp/Identity/EntityFrameworkCore/EfCoreIdentityUserRepository.cs:266]: https://github.com/abpframework/abp/pull/17380

File: modules/identity/src/Volo.Abp.Identity.MongoDB/Volo/Abp/Identity/MongoDB/MongoIdentityUserRepository.cs (hunk 1 above)
Comment (realLiangshiwei) [reply modules/identity/src/Volo.Abp.Identity.MongoDB/Volo/Abp/Identity/MongoDB/MongoIdentityUserRepository.cs:274]: https://github.com/abpframework/abp/pull/17380

Comment (realLiangshiwei) [reply modules/identity/src/Volo.Abp.Identity.MongoDB/Volo/Abp/Identity/MongoDB/MongoIdentityUserRepository.cs:274]: > This won't be efficient.

https://github.com/abpframework/abp/pull/17380

File: modules/identity/src/Volo.Abp.Identity.EntityFrameworkCore/Volo/Abp/Identity/EntityFrameworkCore/EfCoreIdentityUserRepository.cs (hunk 2 above)
Comment (realLiangshiwei) [reply modules/identity/src/Volo.Abp.Identity.EntityFrameworkCore/Volo/Abp/Identity/EntityFrameworkCore/EfCoreIdentityUserRepository.cs:266]: > This method is not performant. Should be revisited.

https://github.com/abpframework/abp/pull/17380

**Pull request merged**